src/lib/operators/table_scan/single_column_table_scan_impl.cpp

```diff
@@ -61,13 +78,79 @@ void SingleColumnTableScanImpl::handle_value_column(const BaseValueColumn& base_
   });
 }
 
-void SingleColumnTableScanImpl::handle_dictionary_column(const BaseDictionaryColumn& base_column,
+void SingleColumnTableScanImpl::handle_dictionary_column(const BaseDeprecatedDictionaryColumn& base_column,
                                                          std::shared_ptr<ColumnVisitableContext> base_context) {
+  _handle_dictionary_column(base_column, base_context);
+
+  // Implementation of an unoptimized scan
+
+  // auto context = std::static_pointer_cast<Context>(base_context);
+  // auto& matches_out = context->_matches_out;
+  // const auto& mapped_chunk_offsets = context->_mapped_chunk_offsets;
+  // const auto chunk_id = context->_chunk_id;
+
+  // const auto left_column_type = _in_table->column_type(_left_column_id);
+
+  // resolve_data_type(left_column_type, [&](auto type) {
+  //   using Type = typename decltype(type)::type;
+
+  //   auto& left_column = static_cast<const DeprecatedDictionaryColumn<Type>&>(base_column);
+
+  //   auto left_column_iterable = create_iterable_from_column(left_column);
+
+  //   auto right_value_iterable = ConstantValueIterable<Type>{_right_value};
+
+  //   left_column_iterable.with_iterators(mapped_chunk_offsets.get(), [&](auto left_it, auto left_end) {
+  //     right_value_iterable.with_iterators([&](auto right_it, auto right_end) {
+  //       with_comparator(_scan_type, [&](auto comparator) {
+  //         _binary_scan(comparator, left_it, left_end, right_it, chunk_id, matches_out);
+  //       });
+  //     });
+  //   });
+  // });
+}
+
+void SingleColumnTableScanImpl::handle_encoded_column(const BaseEncodedColumn& base_column,
+                                                      std::shared_ptr<ColumnVisitableContext> base_context) {
+  if (base_column.encoding_type() == EncodingType::Dictionary) {
+    const auto& left_column = static_cast<const BaseDictionaryColumn&>(base_column);
+    _handle_dictionary_column(left_column, base_context);
+    return;
+  }
+
+  auto context = std::static_pointer_cast<Context>(base_context);
+  auto& matches_out = context->_matches_out;
+  const auto& mapped_chunk_offsets = context->_mapped_chunk_offsets;
+  const auto chunk_id = context->_chunk_id;
+
+  const auto left_column_type = _in_table->column_type(_left_column_id);
+
+  resolve_data_type(left_column_type, [&](auto type) {
+    using Type = typename decltype(type)::type;
+
+    resolve_encoded_column_type<Type>(base_column, [&](const auto& left_column) {
+      auto left_column_iterable = create_iterable_from_column(left_column);
+
+      auto right_value_iterable = ConstantValueIterable<Type>{_right_value};
+
+      left_column_iterable.with_iterators(mapped_chunk_offsets.get(), [&](auto left_it, auto left_end) {
+        right_value_iterable.with_iterators([&](auto right_it, auto right_end) {
+          with_comparator(_scan_type, [&](auto comparator) {
+            _binary_scan(comparator, left_it, left_end, right_it, chunk_id, matches_out);
```

Comment:
This looks like code duplication from the value column?

Reply:
Yes, I updated the TableScan. This part is gone now.